Scheduled List

marbetschar opened the pull request:
Replaces #98, Fixes #46.

Contains only the changes necessary for Schedule List based upon latest master and the feedback provided in #98 by @danrabbit.

This PR should be way easier to review.

![Screenshot from 2020-05-21 19-29-06](https://user-images.githubusercontent.com/392542/82587766-0cc92880-9b9a-11ea-8dfa-978e450f1707.png)

danrabbit commented on src/ListView.vala:
```diff
@@ -144,21 +186,73 @@ public class Tasks.ListView : Gtk.Grid {
+    private void on_tasks_added (Gee.Collection<ECal.Component> tasks, E.Source source) {
         tasks.foreach ((task) => {
-            var task_row = new Tasks.TaskRow.for_component (task, source);
+            var task_row = new Tasks.TaskRow.for_component (task, source, "@colorAccent", this.source == null);
```
Setting the color this way is super overkill, and in the case of the check button it actually does nothing since it is already using accent color by default

marbetschar replied:
Dropped tint_color property in TaskRow.

danrabbit commented on src/Widgets/TaskRow.vala:
```diff
@@ -334,7 +336,11 @@ public class Tasks.TaskRow : Gtk.ListBoxRow {
     public void update_request () {
-        Tasks.Application.set_task_color (source, check);
+        if (is_scheduled_view && tint_color != null) {
+            Tasks.Application.set_task_color_from_string (tint_color, check);
```
This doesn't actually do anything since the only place this is called sets the check to the accent color (which it already is by default)

danrabbit commented on src/ListView.vala:
```diff
@@ -120,14 +144,32 @@ public class Tasks.ListView : Gtk.Grid {
-                (row as Tasks.TaskRow).update_request ();
-            }
-        });
+            Tasks.Application.set_task_color_from_string ("@colorAccent", editable_title);
```
editable_title already has `Granite.STYLE_CLASS_ACCENT`, so this does nothing

danrabbit commented on src/Widgets/ScheduledRow.vala:
```diff
@@ -0,0 +1,63 @@
+
+        var grid = new Gtk.Grid ();
+        grid.column_spacing = 3;
+        grid.margin_start = 4;
```
To get this lined up with the SourceRow it looks like you probably want
```suggestion
        grid.margin_start = 10;
```

marbetschar replied:
Good catch!

danrabbit commented on src/Widgets/ScheduledRow.vala:
```diff
@@ -0,0 +1,63 @@
+        display_name_label.hexpand = true;
+        display_name_label.margin_end = 9;
+
+        status_image = new Gtk.Image ();
+        status_image.pixel_size = 16;
+
+        var spinner = new Gtk.Spinner ();
+        spinner.active = true;
+        spinner.tooltip_text = _("Connecting…");
+
+        status_stack = new Gtk.Stack ();
+        status_stack.add_named (status_image, "image");
+        status_stack.add_named (spinner, "spinner");
```
It doesn't appear that any of this is ever used, so probably shouldn't be here

marbetschar replied:
Fixed.

danrabbit commented on src/Widgets/ScheduledRow.vala:
```diff
@@ -0,0 +1,63 @@
+
+public class Tasks.ScheduledRow : Gtk.ListBoxRow {
+
+    private Gtk.Image icon;
+    private Gtk.Image status_image;
+    private Gtk.Label display_name_label;
+    private Gtk.Stack status_stack;
+    private Gtk.Revealer revealer;
```
These probably don't all need to be declared ahead of time since they're only used in the construct block

marbetschar replied:
Fixed.

marbetschar commented:
@danrabbit thanks for your feedback, good points! I reworked the code a bit and also added a `Gtk.Stack` to switch between the newly created `scheduled_title` and `editable_title`. The scheduled list title now uses the default accent color which this also improves the contrast level.
